src/wolfgame.py

```diff
@@ -2934,27 +2903,38 @@ def fgoat(var, wrapper, message):
     wrapper.send(messages["goat_success"].format(wrapper.source, goatact, togoat))
 
 @handle_error
-def return_to_village(var, chan, target, *, show_message):
+def return_to_village(var, target, *, show_message):
+    # Note: we do not manipulate or check target.disconnected, as that property
+    # is used to determine if they are entirely dc'ed rather than just maybe using
+    # a different account or /parting the channel. If they were dced for real and
+    # rejoined IRC, the join handler already took care of marking them no longer dced.
     with var.GRAVEYARD_LOCK:
-        temp = target.lower()
-        if temp.nick in var.DISCONNECTED:
-            account, hostmask, when, what = var.DISCONNECTED[temp.nick]
-            if ((not var.DISABLE_ACCOUNTS and temp.account is not None and users.equals(temp.account, account)) or
-                (not var.ACCOUNTS_ONLY and temp.userhost is not None and users.equals(temp.userhost, hostmask))):
-
-                del var.DISCONNECTED[temp.nick]
-                var.LAST_SAID_TIME[temp.nick] = datetime.now()
-                for roleset in var.ORIGINAL_ROLES.values():
-                    if "(dced)" + temp.nick in roleset:
-                        roleset.remove("(dced)" + temp.nick)
-                        roleset.add(temp.nick)
-
-                if temp.nick in var.DCED_PLAYERS:
-                    var.PLAYERS[temp.nick] = var.DCED_PLAYERS.pop(temp.nick)
-
-                if show_message:
-                    channels.Main.mode(("+" + hooks.Features["PREFIX"]["+"], target))
-                    channels.Main.send(messages["player_return"].format(target))
+        if target in var.DISCONNECTED:
+            del var.DISCONNECTED[target]
+            var.LAST_SAID_TIME[target.nick] = datetime.now()
+            for roleset in var.ORIGINAL_ROLES.values():
+                if "(dced)" + target.nick in roleset:
+                    roleset.remove("(dced)" + target.nick)
+                    roleset.add(target.nick)
+
+            if target.nick in var.DCED_PLAYERS:
+                var.PLAYERS[target.nick] = var.DCED_PLAYERS.pop(target.nick)
+
+            if show_message:
+                channels.Main.mode(("+" + hooks.Features["PREFIX"]["+"], target))
+                channels.Main.send(messages["player_return"].format(target))
+        else:
+            # this particular user doesn't exist in var.DISCONNECTED, but that doesn't
+            # mean that they aren't dced. They may have rejoined as a different nick,
+            # for example, and we want to mark them as back without requiring them to do
+            # a !swap.
+            if var.ACCOUNTS_ONLY or target.account:
+                userlist = users._get(account=target.account, allow_multiple=True) # FIXME
+            else: # match host (hopefully the ircd uses vhosts to differentiate users)
```

Comment:
This should be `elif not var.ACCOUNTS_ONLY:`

Reply:
Not needed. If ACCOUNTS_ONLY is set, it will always execute the if. Therefore, the else is automatically "not ACCOUNTS_ONLY" without needing to specify.